src/Sylius/Bundle/CoreBundle/Resources/config/app/state_machine.yml

```diff
@@ -121,6 +121,10 @@ winzou_state_machine:
                     on: "complete"
                     do: ["@sylius.callback.order_payment", "updateOrderOnPayment"]
                     args: ["object"]
+                process_order:
+                    on: ["fail", "cancel"]
+                    do: ["@sylius.callback.order_payment", "processOrderPayments"]
+                    args: ["object"]
```

Comment:
Also, these ``callback`` services, will be removed, they do not do any good. ;) It should simply use the payment processor itself.